### ISSUE TITLE
BUG - sev-2 - iOS - Some assistive tech can't get 'into'/'onto' screens where the only actionable items are initially offscreen

### ISSUE THREAD

**Issue by TKDickson** [department-of-veterans-affairs/va-mobile-app#9461]
## What happened?

Screens where the only actionable items are below the fold are inaccessible to keyboard-only navigation and invisible to switch control (when set to automatically rotate through actionable items). (With sufficiently-large font size, "screens where actionable items initial load offscreen" can be a LOT of screens). 

One example is the BSSV letter selection screen - on this screen, a keyboard user would get the 'review letters' back button at the top, and then all of the nav at the bottom of the screen, but would not be able to get 'into' the screen to select which info to include, or generate their letter. [Here's the targets for switch and keyboard](https://drive.google.com/file/d/1QyMVEx8L8Wfnfd34w0Zvaj_BovNPGVgG/view?usp=sharing).

![Image](https://github.com/user-attachments/assets/ab5091df-2daf-478c-9d8a-c97c88eaae5b)

On Android, users can get to the action items further down the screen (even when initially offscreen). On iOS, users can get to action items 'below the fold' with a keyboard, as long as at least one action item is in the 'content area' of the screen when it first loads.

## Specs:

- Device:
- OS:
- User Account:
- Accessibility State: Depending on device, may need to have text resized (to replicate the condition under which there's nothing 'actionable' onscreen at first). 
- Also turn on either switch control, or keyboard-only access,

## Steps to Reproduce

If you haven't navigated the app with a keyboard before - connect any bluetooth keyboard to your device. Then in your device's settings, go to Accessibility, then Keyboards (it's in the physical and motor section), and toggle "Full Keyboard Access" on. You should be able to start using your keyboard at this point - and common commands are helpfully listed onscreen (feel free to remap in the commands section if desired)

Then, use your keyboard navigation to go to Home > Benefits > VA letters and documents > review letters > Benefits summary and service verification. Notice how on iOS you cannot go further on the screen/can't get to content offscreen, if all actionable items initially loaded offscreen. ('actionable' includes the rows in the 'military service information' section, even though those rows are not in fact actionable items)

## Desired behavior

Should be able to navigate to actionable content with keyboard navigation only, no matter if it's initially onscreen or not

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: High Low 
- Frequency: High Low 

- 3 - Low
- 2 - High
- 1 - Critical

## Screen shot(s) and additional information

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  

</details>

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [ ] Labels added (front-end, back-end, global, Health, relevant feature, accessibility, etc)
- [ ] Estimate of 1 added (for front-end tickets)
- [ ] Added either to the relevant feature epic (if found during new feature implementation), or the relevant team's bug epic ([Global](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8302), [Health & Benefits](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8303), [API](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8304), [QART](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8305))

**Comment by alexandec** [comment]
This appears to be resolved in iOS 18. Non-actionable items are focusable with Full Keyboard Access. I can scroll up and down through them using the arrow keys. Here's an example using only the keyboard to navigate:


https://github.com/user-attachments/assets/75b6b3f9-e0d1-466c-ab14-e9e7ec2aaf6f

**Comment by alexandec** [comment]
Okay, I tested with FKA turned on and larger text. This problem still exists with larger text sizes.

**Comment by alexandec** [comment]
Some background info on keyboard navigation:
- It's possible for VoiceOver/TalkBack focus to work fine, but for keyboard navigation the focus may not work correctly on the same elements due to a difference in implementation between screen reader and keyboard navigation
- We can't detect whether the user has FKA (Full Keyboard Access) enabled on iOS so we can't enable a workaround only for FKA
- Any keyboard navigation workarounds must be carefully tested with screen readers to ensure they don't introduce regressions
- See the comments on [#9474](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/9474) for additional investigation and info on keyboard navigation

**Comment by jennb33** [comment]
12/11/2024 - @kellylein @timwright12 @oddballpete @Eallan919 @ATeal @TKDickson The MFS team will be taking this ticket into Sprint 7, (12/14/2024-12/27/2024)

**Comment by TKDickson** [comment]
@matt-guest-wilcore the image files are just going to be lost as part of the zenhub -> github projects transition. I was able to copy it over today because zenhub is still active, but that option will quickly disappear.

video file of the bug from my original writeup: https://github.com/user-attachments/assets/65430b9f-03d4-4d1a-926e-edc7d7b87d16

**Comment by matt-guest-wilcore** [comment]
As mentioned, iOS requires at least one actionable item on-screen at load to access elements further down, unlike Android, which allows navigation to offscreen items. This is just how the platforms work.

To address this, I propose making an initially visible element (ex: title, header) accessible on affected screens (I'll need to document/test all relevant screens). I’ve tested this solution and it works, but before moving forward, I’d like UX and accessibility input.

https://github.com/user-attachments/assets/18ef99f4-76e2-4bb2-80a9-25030be8624f

**Comment by jennb33** [comment]
12/18/2024 - @matt-guest-wilcore is implementing his strategy. He's going to start making some elements accessible that were not accessible before, which will allow deeper discovery.

**Comment by jennb33** [comment]
12/20/2024 - Ticket sent to @donmccaughey for PR review

**Comment by DonMcCaugheyUSDS** [comment]
@matt-guest-wilcore approved

As noted in the PR, we should thoroughly test the affected views as part of QA for the next release.

> Here's a list of affected screens I found:
>
> Health Tab > Messages > Inbox > Select a Message (Some messages have the issue)
> Health Tab > Messages > Folders > Sent > Select a Message (Some messages have the issue)
> Benefits > VA letters and documents > Review Letters > Items 1, 2, 4 & 6 are affected

**Comment by jennb33** [comment]
@matt-guest-wilcore this ticket has been approved by Don and can be readied for the next release (it will go to the App Stores on 1/22/2025)

**Comment by jennb33** [comment]
1/10/2025 - Rolling to Sprint 9 for next release (goes out 1/22/2025; is live in stores on 1/28/2025)

**Comment by jennb33** [comment]
1/23/2025 - the deployment scheduled for 1/22/2025 was cancelled. Rolling this ticket to Sprint 10, for the next deployment (scheduled for 2/5/2025, go-live on 2/11/2025 in the App Stores)

**Comment by matt-guest-wilcore** [comment]
2-25-25 - Like this [comment](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/9461#issuecomment-2356896398) mentions things works fine on iOS 18 (I tested on 18.1) and I can verify my changes are working as expected but if I run the code on a simulator on a previous iOS version (like 17.2) then it's almost like my changes don't exist. Seems to be some inconsistencies in behavior depending on the iOS version.

Still, I think it's better to get these changes in as is as it will help alleviate this issue for users on more recent versions of iOS.
